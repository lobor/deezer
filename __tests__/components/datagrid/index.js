import React from 'react';
import Datagrid from 'components/datagrid';
import renderer from 'react-test-renderer';

const headerDatagrid = [
  {
    title: '',
    key: 'picture',
    type: 'img'
  },
  {
    title: 'Titre',
    key: 'title'
  },
  {
    title: 'Artist',
    key: 'artist'
  }
];

const datas = [
  {
    title: 'Toto',
    artist: 'Titi',
    picture: 'haha'
  }
]

describe('test datagrid button', () => {
  it('test render', () => {
    const component = renderer.create(
      <Datagrid header={headerDatagrid} showResult={datas} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
