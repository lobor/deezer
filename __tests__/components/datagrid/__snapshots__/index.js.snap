// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test datagrid button test render 1`] = `
<table
  className="datagrid"
  onMouseUp={[Function]}
>
  <thead>
    <tr>
      <th
        onClick={[Function]}
        style={
          Object {
            "width": "25px",
          }
        }
      >
        <span
          className={null}
        >
          
        </span>
      </th>
      <th
        className="slider"
        onMouseDown={[Function]}
      />
      <th
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className={null}
        >
          Titre
        </span>
      </th>
      <th
        className="slider"
        onMouseDown={[Function]}
      />
      <th
        onClick={[Function]}
        style={Object {}}
      >
        <span
          className={null}
        >
          Artist
        </span>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <img
          src="haha"
        />
      </td>
      <td
        className="slider"
      />
      <td>
        Toto
      </td>
      <td
        className="slider"
      />
      <td>
        Titi
      </td>
    </tr>
  </tbody>
</table>
`;
